docs/adrs/0009-nimbus-web-sdk-architecture.md

```diff
@@ -0,0 +1,101 @@
+# Nimbus Web SDK Architecture
+
+* Status: proposed
+* Deciders: James Hugman, Charlie Humphreys, Yashika Khurana, Jared Lockhart
+* Date: 2023-15-03
+
+Technical Story: https://mozilla-hub.atlassian.net/browse/EXP-3158
+
+## Context and Problem Statement
+
+This is part of a body of work necessary to support the use of Nimbus within web applications. 
+The current Nimbus SDK is written in such a way that it supports client-oriented experimentation — experiments are downloaded, evaluated, and stored on the client, and a feature store is exposed with the experiment branches applied. 
+In order to support web experimentation, the Nimbus SDK will need to be updated to be stateless, to support a more statically defined set of helper methods, and to have additional support for Python.
+
+Ultimately, the problem we're trying to solve can be boiled down to one question — how can we update the Nimbus SDK to support web applications while continuing to support the existing clients?
+
+## Decision Drivers
+
+* Architecture decisions for Nimbus on the web[<sup>[1.i]</sup>](#links)
+  * The core logic of the existing SDK will be re-used for the Nimbus web service (Cirrus)
+  * The SDK needs to support Python
+  * The SDK needs to be stateless
+
+## Considered Options
+
+* Separate `nimbus-core`, `nimbus-sdk`, and `cirrus-sdk` libraries
+* Cargo features, one library
+* Cargo features, separate library
+* Expose a stateless API alongside the stateful API
```

Comment:
ISTR this is the `Do Nothing` option: compile both Cirrus and Nimbus SDKs for both web apps _and_ mobile apps.

Reply:
Yep – it might be worth me adding that in parens here